clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/AggKafkaConsumerService.java

```diff
@@ -454,4 +454,32 @@ Set<String> getKafkaUrlsFor(PubSubTopic versionTopic) {
     }
     return kafkaUrls;
   }
+
+  String getIngestionInfoFor(PubSubTopic versionTopic, PubSubTopicPartition pubSubTopicPartition) {
+    StringBuilder consumerIngestionInfo = new StringBuilder(
+        String.format("Consumer information for ingestion task of %s with %s\n", versionTopic, pubSubTopicPartition));
+    for (String kafkaUrl: kafkaServerToConsumerServiceMap.keySet()) {
+      if (hasConsumerAssignedFor(kafkaUrl, versionTopic, pubSubTopicPartition)) {
+        AbstractKafkaConsumerService consumerService = getKafkaConsumerService(kafkaUrl);
+        Map<PubSubTopicPartition, TopicPartitionIngestionInfo> topicPartitionIngestionInfoMap =
+            consumerService.getIngestionInfoFromConsumer(versionTopic, pubSubTopicPartition);
+        consumerIngestionInfo.append(String.format("\t\tKafka URL: %s: \n", kafkaUrl));
+        for (PubSubTopicPartition topicPartition: topicPartitionIngestionInfoMap.keySet()) {
+          TopicPartitionIngestionInfo topicPartitionIngestionInfo = topicPartitionIngestionInfoMap.get(topicPartition);
+          consumerIngestionInfo.append(
+              String.format(
```

Comment:
Is the intention of this only for admin tooling?

If so, could we maybe use a json format for this?  It'd mean we could parse this better and more programmatically in case we wanted to render this information elsewhere

Reply:
I think this is only for debugging purpose to get the topic partition related information without doing heap dump and checking logs for stuck or slow ingestion. It is just for print out some readable information for the topic partitions on the interested consumer thread.

Reply:
That sounds reasonable, I can try to change it to json format.